extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -0,0 +1,192 @@
+	}
+
+	return render_player( $track_list, $attributes );
+}
+
+/**
+ * Renders the HTML for the Podcast player and tracklist.
+ *
+ * @param array $track_list the list of podcast tracks.
+ * @param array $attributes Array containing the Podcast Player block attributes.
+ * @return string the HTML for the podcast player.
+ */
+function render_player( $track_list, $attributes ) {
+	global $content_width;
+
+	$player_data = array(
+		'type'         => 'audio',
+		// Don't pass strings to JSON, will be truthy in JS.
+		'tracklist'    => true,
+		'tracknumbers' => true,
+		'images'       => true,
+		'artists'      => true,
+		'tracks'       => $track_list,
+	);
+
+	$outer         = 22; // Default padding and border of wrapper.
+	$default_width = 640;
+	$theme_width   = empty( $content_width ) ? $default_width : ( $content_width - $outer );
+
+	// If there are no tracks (it is possible) then display appropriate user facing error message.
+	if ( empty( $track_list ) ) {
+		return '<p>' . __( 'No tracks available to play.', 'jetpack' ) . '</p>';
+	}
+
+	ob_start();
+	wp_playlist_scripts( 'audio' );
+	/**
+	 * Prints and enqueues playlist scripts, styles, and JavaScript templates.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @param string $type  Type of playlist. Possible values are 'audio' or 'video'.
+	 * @param string $style The 'theme' for the playlist. Core provides 'light' and 'dark'.
+	 */
+	do_action( 'wp_playlist_scripts', 'audio', 'light' );
+
+	$initial_track_src = ! empty( $track_list[0]['src'] ) ? $track_list[0]['src'] : '';
+	$css_class         = Jetpack_Gutenberg::block_classes( BLOG_SLUG, $attributes );
+
+	?>
+	<div class="wp-block-<?php echo esc_attr( $css_class ); ?> wp-playlist wp-audio-playlist wp-playlist-light">
+		<div class="wp-playlist-current-item"></div>
+		<audio src="<?php echo esc_attr( $initial_track_src ); ?>" controls="controls" preload="none" width="<?php echo esc_attr( (int) $theme_width ); ?>"></audio>
+		<div class="wp-playlist-next"></div>
+		<div class="wp-playlist-prev"></div>
+		<noscript>
+			<ol>
+				<?php
+				foreach ( $track_list as $track ) :
+					printf( '<li>%s</li>', esc_url( $track['src'] ) );
+				endforeach;
+				?>
+			</ol>
+		</noscript>
+		<script type="application/json" class="wp-playlist-script"><?php echo wp_json_encode( $player_data ); ?></script>
+	</div>
+	<?php
+	/*
+	* Enqueue necessary scripts and styles.
+	*/
+	Jetpack_Gutenberg::load_assets_as_required( 'podcast-player' );
+
+	return ob_get_clean();
+}
+
+/**
+ * Gets a list of tracks for the supplied RSS feed.
+ *
+ * @param string $feed the RSS feed to load and list tracks for.
+ * @param int    $quantity the number of tracks to return.
+ * @return array|WP_Error the feed's tracks or a error object.
+ */
+function get_track_list( $feed, $quantity = 5 ) {
+	if ( empty( $feed ) ) {
+		return new WP_Error( 'missing_feed', __( 'Podcast audio RSS feed missing.', 'jetpack' ) );
```

Comment:
We're returning `WP_Error` a few times here, with meaningful information about why we have an error, but we don't surface the error messages to the site owner. Maybe we should update `render_block` to return that information to site admins?

Reply:
I think the block already gives information to the user when some of these errors happen:

yeah, maybe. Even we can reuse the error message.

Reply:
![image](https://user-images.githubusercontent.com/77539/76556945-46363700-6479-11ea-8d0b-74f5329aecf3.png)

![image](https://user-images.githubusercontent.com/77539/76557634-b1ccd400-647a-11ea-8ea1-0d78ff71a110.png)

![image](https://user-images.githubusercontent.com/77539/76557542-8d70f780-647a-11ea-89a2-9b94650191fa.png)

Changes [here](2efe21c25bdee1d88ba824dc037034bf817c9cec).